### DISCUSSION TITLE
Nextcloud AIO in latest Unraid 6.12.4 utilizing Cloudflare Tunnels - A difinitive install guide / video needed

### DISCUSSION THREAD

**Discussion by aedavis22** [nextcloud/all-in-one#3700] [Questions] [answered]
I have been trying to get this to work for over a month without success.  I run Unraid 6.12.4 in my homelab and currently use Cloudflare Tunnels for Zero Trust interaction.  I for the life of me cannot get this to spin up properly and give me a Nextcloud login screen.  The containers never fully spin up on the AIO Management screen - see below:
![image](https://github.com/nextcloud/all-in-one/assets/108083924/a72bf0c1-f035-4972-bb4d-06b21b1ad776)

Eventually it will go to a some containers have restarted message but never will it spin up to green.  Apache remains yellow along with Nextcloud.  

I am obviously doing something wrong and could use some assistance with getting this functioning.  I also needed insight on making Talk work as it requires the TURN server port to be exposed and I am unaware of how to manage this using Cloudflare Tunnels.  

Thank you all in advance.  

Tony D.

**Comment by szaimen** [comment, marked as answer]
Hi, first it would be good to debug which of the containers is not starting correctly and restarting.

If that is not an option for you, I would recommend to clean things properly by following https://github.com/nextcloud/all-in-one#how-to-properly-reset-the-instance and then starting freshly with a new instance and following https://github.com/nextcloud/all-in-one/blob/main/reverse-proxy.md#6-how-to-debug-things

**Reply by aedavis22** [reply]
I will go over everything step by step.  I must have missed something.  Mainly utilization of Cloudflare Tunnels and not NGINX or SWAG.  This seems to be the point where alot of us run into the trouble.

**Reply by aedavis22** [reply]
Ok - so I recreated everything in Unraid for the Nextcloud AIO docker.  I first used Cloudflare Tunnels to create a subdomain for the AIO instance and pointed it at the Apache2 on  11000.  Again it hung up while creating containers and so I unchecked the Talk ones (App and recorder) and then it went through and got me to a login!!  Joy!!  After that I added them back and again, it hung.  I then researched the link that you provided and saw that the 3478 port could be conflicting in which it was with something called proxy-net.  I changed the port using the environment flag to 3479 and it created and spun up.  The problem now is when I set up a test call, I cannot join with video from a device outside of my local lan.  It seems to be a issue with the AIO version as I have it set up using the standard Nextcloud docker in Unraid and the video calls connect fine.  Perhaps something to do with the high performance enhancements??

Any help or direction you can give will help.  I am wondering if port 3479 is not being exposed via Cloudflare?  How would it know to do it I mean?

Thank you in advance.

A.

**Reply by szaimen** [reply]
Hi, see point 9 here: https://github.com/nextcloud/all-in-one#notes-on-cloudflare-proxytunnel

**Reply by aedavis22** [reply]
I  could not find the point that you were referencing above.. I  went ahead and disabled the high performance backend and video calls now work properly from different networks i.e. from a meeting link I send out to someone outside of my network.  All of that is  working ok.  My latest weirdness is the Borg Backup  system.  I want to test it but I cannot do that without destroying my current working version.  I am getting weird messages from AppData Backup about all of the AIO containers (Debug log 89ff1dbb-5e1e-4161-93c0-10d72b0f3baa) and because of that I fear that the AppData backup is not functioning for NextCloud AIO. I only have Borg to rely on and what I am wondering is how can I test functionality without destroying my setup?  Is there a way to restore to a dummy docker master container with a slightly different name?  Please advise.

Thank you!!

**Reply by szaimen** [reply]
> I could not find the point that you were referencing above.

I meant this point:
![image](https://github.com/nextcloud/all-in-one/assets/42591237/01a01367-e732-472f-9b9c-3f9689ef362d)

> I want to test it but I cannot do that without destroying my current working version.

What exactly is destroyed? It is expected that starting the backup first stops all containers. However you can start them afterwards again. Is this not working? Can you post all logs and possibly a screencast of what you are seeing?

**Reply by aedavis22** [reply]
Ok will do regarding the turn server.  

On the Borg test... The only way to be sure if it is working properly is to remove uninstall Nextcloud AIO and then restore from the backup.  Once removed, if something doesn't restore properly, that leaves me with a broken setup yes?  Keep in mind this is me asking - not stating.  You are the expert on this and I am seeking your help and guidance.  I do appreciate it.  Also - is there a way to send the local Borg data elsewhere after as an automated process to get it offsite?  Thank you

**Reply by szaimen** [reply]
> The only way to be sure if it is working properly is to remove uninstall Nextcloud AIO and then restore from the backup. Once removed, if something doesn't restore properly, that leaves me with a broken setup yes?

Ah that is what you mean. In general you can also simply use the restore feature that is in the AIO interface in the current state without destroying the setup fiest. It will use the same restore logic so is a good test already.



> Also - is there a way to send the local Borg data elsewhere after as an automated process to get it offsite?

See https://github.com/nextcloud/all-in-one#are-remote-borg-backups-supported
